### PR TITLE
s3 - add state option in bucket encryption filter

### PR DESCRIPTION
**Pull request by thisisshi**
Resolves https://github.com/capitalone/cloud-custodian/issues/2401

### CODE REVIEW

File: tests/test_s3.py [hunk 1]
```diff
@@ -207,6 +207,76 @@ def test_s3_bucket_encryption_filter_kms(self):
...
+                }
+            ]
+            }, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertRaises(ClientError, client.get_bucket_encryption, Bucket=bname)
+
+
+    def test_s3_filter_bucket_encryption_enabled(self):
+        bname = 'c7n-bucket-with-default-encryption'
```
Comment (kapilt) [inline tests/test_s3.py]: this case is already tested in other tests.

File: c7n/resources/s3.py [hunk 2]
```diff
@@ -2795,8 +2802,14 @@ def process_bucket(self, b):
...
-        for sse in rules:
-            if self.filter_bucket(b, sse):
+        # default `state` to True as previous impl assumed state == True
+        # to preserve backwards compatibility
+        if self.data.get('state', True):
+            for sse in rules:
+                if self.filter_bucket(b, sse):
+                    return True
+        else:
+            if not rules:
```
Comment (kapilt) [inline c7n/resources/s3.py]: this isn't quite right, the absence of rules doesn't quite mesh with the ability to filter on particular rule expressions with a negation/state=false.

ie. a policy with

```
   crypto: aws:kms
   state: false
```
would ignore buckets with AES256.

**Comment by thisisshi** [comment]
@abhilashjavvaji are you running off of master? the latest release has not been released to include this just yet, you'll need to pull from git to run this.

**Comment by abhilashjavvaji** [comment]
Oh okay, I thought it is released,, so I should clone the repo and use that code to get this working ?

**Comment by thisisshi** [comment]
Yes, follow the realms for the dev install

**Comment by abhilashjavvaji** [comment]
Thanks for quick response,, do you know if this is getting released anytime soon ? So that I don't have to change the way I am doing today for a short amount of time!

**Comment by thisisshi** [comment]
we're due for a new release soon, I'm not sure when exactly it will be though

**Pull request merged**